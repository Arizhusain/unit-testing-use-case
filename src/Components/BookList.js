import React, { Component } from "react";
import Book from "./Book";

class BookList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      bookDetailsObj: {
        id: '',
        bookName: '',
        bookAuthor: '',
      },
      bookInfo: [
        {
          id: 1,
          bookName: "The Intelligent Investor",
          bookAuthor: "Benjamin Graham",
        },
      ],
    };
  }
  render() {
    const handleChange = (event) => {
      const lastElem = this.state.bookInfo.slice(-1)[0];
      this.setState({
        bookDetailsObj: {
          ...this.state.bookDetailsObj,
          [event.target.name]: event.target.value,
          id: lastElem.id + 1,
        },
      });
    };
    const addBookDetails = (event) => {
      event.preventDefault();
      if (
        this.state.bookInfo.find(
          (item) => item.bookName === this.state.bookDetailsObj.bookName
        )
      ) {
        alert("Book Already present, please change thebook name");
      } else {
        this.setState((prevState) => ({
          bookInfo: [...prevState.bookInfo, this.state.bookDetailsObj],
        }));
      }
    };

    return (
      <div>
        <h1>Welcome to My Favorite Book Portal</h1>
        <hr />
        <h2>
          {this.props.name} - {this.props.numberOfBooks}
        </h2>
        <hr />
        <h4>Add book details for reading list</h4>
        Book Name :{" "}
        <input
          name="bookName"
          type="text"
          placeholder="Book Name"
          value={this.state.bookDetailsObj.bookName}
          onChange={handleChange}
          required
          autoComplete="off"
        />
        <br />
        Book Author :{" "}
        <input
          name="bookAuthor"
          type="text"
          placeholder="Author"
          value={this.state.bookDetailsObj.bookAuthor}
          onChange={handleChange}
          required
          autoComplete="off"
        />
        <br />
        <hr />
        <Book bookInfo={this.state.bookInfo} addBookDetails={addBookDetails} />
      </div>
    );
  }
}

export default BookList;
