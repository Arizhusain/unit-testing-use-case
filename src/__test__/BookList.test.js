import { render, screen } from "@testing-library/react"
import BookList from "../Components/BookList";
import renderer from 'react-test-renderer';

//5. Use decribe block and it/test block appropriately
describe('BookListComponent', () => {
    //3. Have a right BookList component snapshot
      it('has right BookList component snapshot', () => {
        const snapshotJSON = renderer.create(<BookList name='Book Library' numberOfBooks='1000'/>).toJSON();
        expect(snapshotJSON).toMatchSnapshot();
      });
    //4. Test whether BookList component has the following text in it
      it('should have Welcome to My Favorite Book Portal in the document', () => {
        render(<BookList/>)
        const elem = screen.getByText('Welcome to My Favorite Book Portal');
        expect(elem).toBeInTheDocument();
      })
      
})
