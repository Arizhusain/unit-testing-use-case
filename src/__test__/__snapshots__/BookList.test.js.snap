// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookListComponent has right BookList component snapshot 1`] = `
<div>
  <h1>
    Welcome to My Favorite Book Portal
  </h1>
  <hr />
  <h2>
    Book Library
     - 
    1000
  </h2>
  <hr />
  <h4>
    Add book details for reading list
  </h4>
  Book Name :
   
  <input
    autoComplete="off"
    name="bookName"
    onChange={[Function]}
    placeholder="Book Name"
    required={true}
    type="text"
    value=""
  />
  <br />
  Book Author :
   
  <input
    autoComplete="off"
    name="bookAuthor"
    onChange={[Function]}
    placeholder="Author"
    required={true}
    type="text"
    value=""
  />
  <br />
  <hr />
  <div>
    <button
      onClick={[Function]}
    >
      Add to reading list
    </button>
    <br />
    <h2>
      Reading list : 
    </h2>
    <ul>
      <li
        data-testid="1"
      >
        The Intelligent Investor
         by 
        Benjamin Graham
      </li>
    </ul>
  </div>
</div>
`;
